src/core/thread/mle.cpp

```diff
@@ -2424,9 +2419,17 @@ ThreadError Mle::HandleLeaderData(const Message &aMessage, const Ip6::MessageInf
     }
 
     // Network Data
-    mNetif.GetNetworkDataLeader().SetNetworkData(leaderData.GetDataVersion(), leaderData.GetStableDataVersion(),
-                                                 (mDeviceMode & ModeTlv::kModeFullNetworkData) == 0,
-                                                 networkData.GetNetworkData(), networkData.GetLength());
+    if (Tlv::GetTlv(aMessage, Tlv::kNetworkData, sizeof(networkData), networkData) == kThreadError_None)
+    {
+        VerifyOrExit(networkData.IsValid(), error = kThreadError_Parse);
+
+        // Ignore Network Data which doesn't have newer version
+        VerifyOrExit(diff > 0);
```

Comment:
I'm not sure we can assume that the Network Data Version changes when the Partition ID also changes.